cgp/ea/mu_plus_lambda.py

```diff
@@ -167,41 +169,57 @@ def _create_new_offspring_generation(self, pop: Population) -> List[IndividualBa
 
     def _compute_fitness(
-        self, combined: List[IndividualBase], objective: Callable[[IndividualBase], IndividualBase]
+        self,
+        combined: List[IndividualBase],
+        objective: Union[
+            Callable[[IndividualBase], IndividualBase],
+            List[Callable[[IndividualBase], IndividualBase]],
+        ],
+        ignore_hurdles=False,
     ) -> List[IndividualBase]:
+        def compute_fitness_hurdle(ind_evaluating: List[IndividualBase]) -> float:
+            return np.percentile(
+                np.unique([ind.fitness_current_objective for ind in ind_evaluating]),
+                self.hurdle_percentile[ind_evaluating[0].objective_idx] * 100,
+            )
 
         self.update_n_objective_calls(combined)
 
-        # computes fitness on all individuals, objective functions
-        # should return immediately if fitness is not None
-        if self.n_processes == 1:
-            # don't use the process pool if running just a single
-            # process to avoid any multiprocessing-associated overhead
-            combined = list(map(objective, combined))
-        else:
-            assert isinstance(self.process_pool, mp.pool.Pool)
-            combined = self.process_pool.map(objective, combined)
+        if callable(objective):
+            objective = [objective]
 
-        return combined
-
-    def _sort(self, combined: List[IndividualBase]) -> List[IndividualBase]:
-        def sort_func(ind: IndividualBase) -> float:
-            """Return fitness of an individual, return -infinity for an individual
-            with fitness equal nan, or raise error if the fitness is
-            not a float.
+        ind_evaluating = list(combined)
+        ind_done_evaluating = []
+        for obj_idx, obj in enumerate(objective):
 
-            """
-            if np.isnan(ind.fitness):
-                return -np.inf
+            # prime individuals for receiving fitness for this objective
+            for ind in ind_evaluating:
+                ind.objective_idx = obj_idx
 
-            if isinstance(ind.fitness, float):
-                return ind.fitness
+            if self.n_processes == 1:
+                # don't use the process pool if running just a single
+                # process to avoid any multiprocessing-associated overhead
+                ind_evaluating = list(map(obj, ind_evaluating))
             else:
-                raise ValueError(
-                    f"IndividualBase fitness value is of wrong type {type(ind.fitness)}."
-                )
+                assert isinstance(self.process_pool, mp.pool.Pool)
+                ind_evaluating = self.process_pool.map(obj, ind_evaluating)
 
-        return sorted(combined, key=sort_func, reverse=True)
+            fitness_hurdle = compute_fitness_hurdle(ind_evaluating)
+
+            ind_evaluating_new = []
+            for ind in ind_evaluating:
+                assert isinstance(ind.fitness_current_objective, float)
```

Comment:
If I understand correctly this will raise an error if an `ind.fitness_current_objective = np.nan` - do we really want that? This can happen easily if one uses eg. division operator genes - should the run always stop then?

Reply:
nope, this wouldn't raise an error:
```python
In [3]: isinstance(np.nan, float)
Out[3]: True
```